__tests__/fetchers.js

```diff
@@ -206,3 +206,34 @@ test('TarballFetcher.fetch properly stores tarball of scoped package in offline
   const exists = await fs.exists(path.join(offlineMirrorDir, '@exponent-configurator-1.0.2.tgz'));
   expect(exists).toBe(true);
 });
+
+test.only('TarballFetcher.fetch retry on 500', async () => {
+  // jest.useRealTimers();
+  const dir = await mkdir('tarball-fetcher');
+  jest.mock('request', factory => options => {
+    return {
+      on: (name, cb) => {
+        if (name === 'response') {
```

Comment:
hmm, thanks for pointing it out. `__tests__/__mocks__/request.js` doesn't have any support for testing stream. I'm trying to figure out a way.